analyses/rna-seq-expression-summary-stats/README.md

```diff
@@ -0,0 +1,181 @@
+## Calculate TPM summary statistics within each cancer group and cohort
+
+**Module authors:** Yuanchao Zhang ([@logstar](https://github.com/logstar))
+
+### Purpose
+
+Within each cancer group and cohort, calculate TPM means, standard deviations, z-scores, and ranks.
+
+### Methods
+
+Select independent RNA-seq samples using `independent-specimens.rnaseq.primary.tsv` in the results of the `independent-samples` analysis module.
+
+Combine CBTN and PNOC into one cohort, PBTA.
+
+For each `cancer_group`, select one of the following two sets of samples:
+
+- Samples from all `cohort`s, e.g. PBTA and GMKF.
+- Samples from each individual `cohort`.
+
+If >= 5 samples are selected, generate the following summary statistics:
+
+- TPM means of each gene across all selected samples, and denote this vector as `mean_TPM_vector`.
+- TPM standard deviations of each gene across all selected samples.
+- z-scores of each gene across all genes, computed as `z_score_vector = (mean_TPM_vector - mean(mean_TPM_vector)) / sd(mean_TPM_vector)`. Call these z-scores as `mean_tpm_cancer_group_wise_z_scores` in the filenames.
+- Ranks of mean TPM of each gene across all selected samples, which takes one of the following four values: `Highest expressed 25%`, `Expression between upper quartile and median`, `Expression between median and lower quartile`, and `Lowest expressed 25%`. If multiple genes have the same mean TPM value, their tied rank is the lowest rank, in order to be conservative on the description of their expression levels.
+
+Combine each type of the summary statistics vectors into a table, with rows as genes, and columns as `cancer_group_cohort`.
+
+Denote the (`n_genes`, `n_cancer_groups`/`n_cancer_group_cohorts`) `mean_TPM_vector` combined matrix as `mean_TPM_matrix`.
+
+Generate z-scores across all `cancer_groups`/`cancer_group_cohorts` as `z_score_matrix = (mean_TPM_matrix - rowMeans(mean_TPM_matrix)) / rowSD(mean_TPM_matrix)`. Call these z-scores as `mean_tpm_gene_wise_z_scores` in the filenames.
+
+Call aforementioned (`n_genes`, `n_cancer_groups`/`n_cancer_group_cohorts`) tables as wide tables.
```

Comment:
It is possible we can remove the wide version of the tables in the future, but we can leave in for now and make an update later if the long tables look to be what is desired.